lib/perso.ml

```diff
@@ -1680,6 +1680,23 @@ and eval_simple_str_var conf base env (_, p_auth) = function
                 |> safe_val
               else null_val
           | _ -> raise Not_found))
+  | "marriage_place_raw" -> (
+      match get_env "fam" env with
+      | Vfam (_, fam, _, m_auth) when mode_local env ->
+          if m_auth then
+            get_marriage_place fam |> sou base
+            |> Util.raw_string_of_place conf
+            |> str_val
+          else null_val
+      | _ -> (
+          match get_env "fam_link" env with
+          | Vfam (_, fam, _, m_auth) ->
+              if m_auth then
+                get_marriage_place fam |> sou base
+                |> Util.raw_string_of_place conf
+                |> str_val
+              else null_val
```

Comment:
we can factorize this if block

Reply:
I dont see how. `fam` and `m_auth` must be known and are different in each block.

Reply:
maybe like this?

```ocaml
      match get_env "fam" env,  get_env "fam_link" env with
      | (Vfam (_, fam, _, m_auth) when mode_local env, _) | ( _ , Vfam (_, fam, _, m_auth)) ->
          if m_auth then
            get_marriage_place fam |> sou base
            |> Util.raw_string_of_place conf
            |> str_val
          else null_val
     | _ -> raise Not_found)
```

Reply:
but mode_local is just:
```ocaml
 let mode_local env =                                                                                
   match get_env "fam_link" env with Vfam _ -> false | _ -> true
```
So the `when mode_local env` in the match is a little weird.
I don't understand what this code is supposed to do and why.
What are "fam" and "fam_link" in `env`?

Reply:
Ok, I think "fam_link" is related to the "Lien Inter Arbre" feature of Geneanet...

Reply:
I don't know, let's keep both of them for now

Reply:
J'arrive pas à faire le 
```
      match get_env "fam" env, get_env "fam_link" env with
      | (Vfam (_, fam, _, m_auth) when mode_local env, _)
      | ( _ , Vfam (_, fam, _, m_auth)) ->
```
Il me dit : 
```
File "lib/perso.ml", line 1685, characters 34-38:
Error: Syntax error: ')' expected
File "lib/perso.ml", line 1685, characters 8-9:
  This '(' might be unmatched

```
!!!
j'ai essayé toutes sortes de parenthèses sans succès!

Reply:
I tried too, I think it is because of the `when mode_local env`

let's just leave it like that, sorry for bothering you